### PR TITLE
feat: add "Mark all as read" button

### PR DESCRIPTION
**Pull request by joebrock666**
https://trello.com/c/2mspnmft

### CODE REVIEW

File: src/views/Chats.vue [hunk 1]
```diff
@@ -117,6 +138,10 @@ export default {
     },
     userId () {
       return this.$store.state.address
+    },
+    unreadMessagesCount() {
+      const messages = this.$store.getters['chat/unreadMessages']
```
Comment (bludnic) [inline src/views/Chats.vue:143]: Use `chat/totalNumOfNewMessages` getter.
It returns a number by default.

File: src/views/Chats.vue [hunk 2]
```diff
@@ -12,25 +12,46 @@
...
             >
-              <template #prepend>
-                <v-icon
-                  :class="`${className}__icon`"
-                  icon="mdi-message-outline"
-                  size="16"
-                />
-              </template>
+              <v-list-item
+                v-if="unreadMessagesCount >= 1"
```
Comment (bludnic) [inline src/views/Chats.vue:21]:
```suggestion
                v-if="unreadMessagesCount > 0"
```

File: src/views/Chats.vue [hunk 3]
```diff
@@ -12,25 +12,46 @@
...
+                :class="`${className}__item ml-2`"
+                @click="readAll"
```
Comment (bludnic) [inline src/views/Chats.vue:23]:
```suggestion
                @click="markAllAsRead"
```

File: src/views/Chats.vue [hunk 4]
```diff
@@ -12,25 +12,46 @@
...
+              >
+                <template #prepend>
+                  <v-icon
+                    :class="`${className}__icon`"
+                    icon="mdi mdi-check-all"
+                    size="20"
```
Comment (bludnic) [inline src/views/Chats.vue:29]: use same icon size as the "Start new chat" icon button

**Review by bludnic — CHANGES_REQUESTED** [review]

File: src/views/Chats.vue (hunk 1 above)
Comment (bludnic) [reply src/views/Chats.vue]: remove local `const messages`, return the getter

File: src/assets/styles/themes/adamant/_buttons.scss [hunk 5]
```diff
@@ -9,6 +9,10 @@
   text-transform: unset;
 }
 
+.round-btn {
```
Comment (bludnic) [inline src/assets/styles/themes/adamant/_buttons.scss:12]: Remove this class. Icon buttons are rounded by default. Check Vuetify docs.

**Review by bludnic — CHANGES_REQUESTED** [review]

File: src/assets/styles/themes/adamant/_buttons.scss (hunk 5 above)
Comment (joebrock666) [reply src/assets/styles/themes/adamant/_buttons.scss:12]: I don't see how it's true. I tried using v-btn with v-icon in one tag but it's not rounded as in examples in docs. It's square with some border-radius.

Comment (bludnic) [reply src/assets/styles/themes/adamant/_buttons.scss:12]: some app styles may affected the Vuetify styles

File: src/views/Chats.vue [hunk 6]
```diff
@@ -11,29 +11,40 @@
...
-                  size="16"
-                />
-              </template>
+            <v-row no-gutters>
+              <v-btn
+                :class="`${className}__btn`"
+                @click="markAllAsRead"
+                variant="plain"
+                v-if="unreadMessagesCount > 0"
+                icon="mdi mdi-check-all"
```
Comment (bludnic) [inline src/views/Chats.vue]:
```suggestion
                icon="mdi-check-all"
```

**Review by bludnic — CHANGES_REQUESTED** [review]

File: src/components/PartnerInfo.vue [hunk 7]
```diff
@@ -12,15 +12,11 @@
...
-          class="close-icon round-btn"
+          icon="mdi-close"
+          class="close-icon"
           :size="36"
           @click="show = false"
         >
-          <v-icon
-            icon="mdi-close"
-            :size="24"
-          />
```
[the remark this answers is not in the document]
Comment (joebrock666) [reply src/components/PartnerInfo.vue:23]: How to maintain the needed border-radius for single icons if i rollback though?

[the remark this answers is not in the document]
Comment (bludnic) [reply src/components/PartnerInfo.vue:23]: image

The button is still rounded when passing icon as a slot. 

image
image

File: src/assets/styles/generic/_vuetify.scss [hunk 8]
```diff
@@ -36,7 +36,6 @@ html {
 
 .v-btn {
   letter-spacing: normal;
-  border-radius: 2px;
```
[the remark this answers is not in the document]
Comment (bludnic) [reply src/assets/styles/generic/_vuetify.scss:39]: Don't remove `border-radius`. Just fix the selector. Apply that styles only for button but not for icon buttons.

**Review by bludnic — CHANGES_REQUESTED** [review]

File: src/components/PartnerInfo.vue (hunk 7 above)
[the remark this answers is not in the document]
Comment (joebrock666) [reply src/components/PartnerInfo.vue:23]: Oh, I didn't realize it, now I get it

File: src/components/PartnerInfo.vue [hunk 9]
```diff
@@ -14,7 +14,6 @@
           variant="text"
           icon
           class="close-icon"
-          :size="36"
```
Comment (bludnic) [inline src/components/PartnerInfo.vue:17]: rollback size

File: src/components/WalletCard.vue [hunk 10]
```diff
@@ -20,10 +20,10 @@
 
         <template #append>
           <v-btn
-            icon
```
Comment (bludnic) [inline src/components/WalletCard.vue:23]: rollback

File: src/views/Chats.vue [hunk 11]
```diff
@@ -11,29 +11,41 @@
...
+                  size="small"
+                  variant="plain"
+                >
+                </v-btn>
+              <v-spacer></v-spacer>
+              <v-btn
+                :class="`${className}__item`"
+                @click="showChatStartDialog = true"
+                variant="plain"
+              >
```
Comment (bludnic) [inline src/views/Chats.vue:31]: prettier

File: src/views/Chats.vue [hunk 12]
```diff
@@ -224,6 +242,10 @@ export default {
   }
 }
 
+.v-row {
```
Comment (bludnic) [inline src/views/Chats.vue:245]:
```suggestion
&__chats-actions {
```

**Review by bludnic — CHANGES_REQUESTED** [review]

File: src/components/PartnerInfo.vue (hunk 9 above)
Comment (joebrock666) [reply src/components/PartnerInfo.vue:17]: wdym? the size prop is in v-icon tag

Comment (bludnic) [reply src/components/PartnerInfo.vue:17]: this is the of the button not of the icon

File: src/views/Chats.vue [hunk 13]
```diff
@@ -1,45 +1,38 @@
...
-
-            <transition-group
-              v-if="isFulfilled"
-              name="messages"
-            >
-              <template
-                v-for="transaction in messages"
-                :key="transaction.contactId"
+          <v-list subheader class="pa-0" bg-color="transparent" v-if="isFulfilled">
+            <v-row class="chats-view__chats-actions" no-gutters>
```
Comment (bludnic) [inline src/views/Chats.vue:7]:
```suggestion
            <v-row class="`${className}__chats-actions`" no-gutters>
```

**Review by bludnic — CHANGES_REQUESTED** [review]

File: src/views/Chats.vue [hunk 14]
```diff
@@ -1,45 +1,38 @@
...
+            <v-row :class="`${className}__chats-actions`" no-gutters>
+              <v-btn
+                :class="`${className}__btn mt-2`"
+                @click="markAllAsRead"
+                v-if="unreadMessagesCount > 0"
+                icon="mdi-check-all"
+                size="small"
+                variant="plain"
               >
+              </v-btn>
```
Comment (bludnic) [inline src/views/Chats.vue:16]: closing tag is optional

File: src/views/Chats.vue [hunk 15]
```diff
@@ -1,45 +1,38 @@
...
+              <v-spacer></v-spacer>
```
Comment (bludnic) [inline src/views/Chats.vue:17]:
```suggestion
              <v-spacer/>
```

**Review by bludnic — CHANGES_REQUESTED** [review]

**Review by bludnic — APPROVED** [review]

**Pull request merged**